app/controllers/spree/admin/shipping_methods_controller_decorator.rb

```diff
@@ -29,21 +28,8 @@ def collection
         collection
       end
 
-      # This method was originally written because ProductDistributions referenced shipping
-      # methods, and deleting a referenced shipping method would break all the reports that
-      # queried it.
-      # This has changed, and now all we're protecting is Orders, which is a spree resource.
-      # Do we really need to protect it ourselves? Does spree do this, or provide some means
-      # of preserving the shipping method information for past orders?
```

Comment:
Yes, it does by implementing soft deletes: marking the row as deleted through an update instead of actually getting rid of it.